### PR TITLE
Refactor with CommonDelayableHook and refactor class Utils

### PR DESCRIPTION
**Pull request by ketal178**
# Refactor using CommonDelayableHook

## 介绍 / Description
Refactor using CommonDelayableHook
## 更改内容 / Types of Changes
- 增强优化 Enhancement/optimization

## 由这个PR解决或关闭的Issues / Issues Fixed or Closed by This PR

-  

## 检查列表 / Checklist

- [x] 我的代码符合本仓库的代码规范 My code follows the code style of this project
- [x] 我已经测试了这些更改，它们可以正常工作，既不会破坏原有任何功能(或我可以解决)，也不会影响对旧版本的支持 I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] 我已合并了对后续工作无意义的commit并确认不会对后续维护造成困扰 I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

**Review by singleNeuron — APPROVED** [review]

**Review by NextAlone — APPROVED** [review]

**Review by NextAlone — DISMISSED** [review]

File: app/src/main/java/me/kyuubiran/hook/RemovePokeGrayTips.kt [hunk 1]
```diff
@@ -18,31 +18,18 @@
...
-import nil.nadph.qnotified.hook.BaseDelayableHook
-import nil.nadph.qnotified.step.Step
+import nil.nadph.qnotified.hook.CommonDelayableHook
 import nil.nadph.qnotified.util.LicenseStatus
-import nil.nadph.qnotified.util.Utils
 
 //屏蔽戳一戳灰字提示
-object RemovePokeGrayTips : BaseDelayableHook() {
-    private const val kr_remove_poke_tips: String = "kr_remove_poke_tips"
+object RemovePokeGrayTips : CommonDelayableHook("kr_remove_poke_tips") {
```
Comment (cinit) [inline app/src/main/java/me/kyuubiran/hook/RemovePokeGrayTips.kt:29]: SyncUtils.PROC_ANY

File: app/src/main/java/me/kyuubiran/hook/RemoveRedDot.kt [hunk 2]
```diff
@@ -20,25 +20,18 @@ package me.kyuubiran.hook
...
 import nil.nadph.qnotified.util.LicenseStatus
 import nil.nadph.qnotified.util.Utils
 import java.lang.reflect.Method
 
 
 //移除小红点
-object RemoveRedDot : BaseDelayableHook() {
-    private const val kr_remove_red_dot: String = "kr_remove_red_dot"
-    var isInit = false
+object RemoveRedDot : CommonDelayableHook("kr_remove_red_dot") {
```
Comment (cinit) [inline app/src/main/java/me/kyuubiran/hook/RemoveRedDot.kt:34]: SyncUtils.PROC_ANY

**Review by cinit — CHANGES_REQUESTED** [review]

**Review by NextAlone — APPROVED** [review]

**Review by cinit — APPROVED** [review]

**Comment by singleNeuron** [comment]
# 重构

## 介绍 / Description

- 拆分Utils.java
- 彻底隔离引导部分，防止第三方包污染
- 由内核统一提供HostInformationProvider，降低性能损失


## 更改内容 / Types of Changes

- 修复 Bugfix
- 增强优化 Enhancement/optimization
- 内核 Core

## 检查列表 / Checklist





- [x] 我的代码符合本仓库的代码规范 My code follows the code style of this project
- [ ] 我已经测试了这些更改，它们可以正常工作，既不会破坏原有任何功能(或我可以解决)，也不会影响对旧版本的支持 I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] 我已合并了对后续工作无意义的commit并确认不会对后续维护造成困扰 I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

**Review by singleNeuron — APPROVED** [review]

**Review by singleNeuron — DISMISSED** [review]

**Review by qwq233 — DISMISSED** [review]

**Review by cinit — DISMISSED** [review]

File: app/src/main/java/nil/nadph/qnotified/activity/FriendSelectActivity.java [hunk 3]
```diff
@@ -175,10 +191,12 @@ public void onClick(View v) {
...
             try {
                 ConfigManager cfg = ExfriendManager.getCurrent().getConfig();
-                if (mActionInt == FriendSelectActivity.class) {
-                    cfg.putString(ConfigItems.qn_chat_tail_friends, ret);
-                    cfg.save();
-                }
+                cfg.putString(targetKeyName, ret);
+                cfg.save();
+//                }     if (mActionInt == FriendSelectActivity.class) {
+//                    cfg.putString(ConfigItems.qn_chat_tail_friends, ret);
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/activity/FriendSelectActivity.java:197]: delete pls

**Review by qwq233 — APPROVED** [review]

File: app/src/main/java/nil/nadph/qnotified/startup/StartupRoutine.java [hunk 4]
```diff
@@ -43,16 +46,16 @@ private StartupRoutine() {
...
             Natives.load(ctx);
         } catch (Throwable e3) {
             log(e3);
         }
-        checkLogFlag();
-        HostInformationProviderKt.init(ReflexUtil.getApplication());
         if (getBuildTimestamp() < 0) {
             return;
         }
+        HostInformationProviderKt.init((Application) ctx);
```
Comment (singleNeuron) [inline app/src/main/java/nil/nadph/qnotified/startup/StartupRoutine.java:59]: HostInformationProvider的初始化必须在getBuildTimestamp之前被执行

**Review by singleNeuron — CHANGES_REQUESTED** [review]

**Review by singleNeuron — APPROVED** [review]

**Review by cinit — APPROVED** [review]

**Review by qwq233 — APPROVED** [review]

**Pull request merged**